app/controllers/users_controller.rb

```diff
@@ -31,22 +32,34 @@ def edit
   # POST /users.json
   def create
     @user = User.new(user_params)
-    user = @user
     @user.profile_picture.attach(user_params[:profile_picture])
     if @user.username === ""
       @user.username = @user.name + rand((User.all.length)..(User.all.length * 100000000)).to_s
```

Comment:
What happens if the username we assign here has already been picked?

Reply:
If the username has already been picked, then, the user stays with the username that he has already picked, this is just for preventing an error when creating the user, as the user cannot be saved without the username, but if the user doesn't pick any username, when registering, then a random unique username is made, based on the user name he has entered in the form, and a random number, just as in twitter. I didn't put any else statement, because i didn't see it necessary.

Reply:
Let's say that user A joins retwitter and we assign them a username of `A1`, then user AA joins retwitter and we asssign them a user name of `A1` here. We end up with two users who have the same username because we're not checking if we already have another user with that username in the database.

Reply:
If the user AA joins retwitter, then, the result username is going to be AA1. And, actually we are checking if the username is unique, as in the database i added the unique: true condition, so the username is not going to be saved if that username already exists.

Reply:
So, if the username is not unique, when we try to save the username here, this will throw an error, right? We should check that the username doesn't exist already, and if it does, then we reassign a different username.